jena-arq/src/main/java/org/apache/jena/sparql/engine/binding/BindingBase.java

```diff
@@ -202,4 +202,19 @@ public static int hashCode(Binding bind) {
         }
         return hash;
     }
+
+    @Override
+    public Binding detach() {
+        Binding newParent = parent == null ? null : parent.detach();
+        Binding result = newParent == parent
+                ? detachWithOriginalParent()
+                : detachWithNewParent(newParent);
+        return result;
+    }
+
+    protected Binding detachWithOriginalParent() {
+        return this;
+    }
+
+    protected abstract Binding detachWithNewParent(Binding newParent);
```

Comment:
This would be nicer in `BindingFactory` ... if we were using Java21 and could use switch patterns. Oh well, another time.

Reply:
I was also thinking that BindingFactory could take care of making bindings independent. But then again, why should BindingFactory have special logic for BindingTDB instances? In principle, `BindingTDB.detach` could be implemented as just setting filling the already present (but unused) cache - so no copy would be needed then.